### ISSUE TITLE
Discussion: renaming "block" to avoid conflation with other services

### ISSUE THREAD

**Issue by cinnamon-bun** [fraction/oasis#134]
(crossposted from ssb)

Might we choose a different word for our blocking functionality?

I’m seeing lots of assumptions about blocking amongst new users, leading them to react strongly when blocked:
* Does it have an aggressive connotation? (no)
* If I’m blocked by one person am I made invisible to everyone? (no)
* Is it bidirectional (blocking someone means they can’t see you either)? (no)

On Twitter and Mastodon, “blocking” feels like a strong/aggressive action associated with “cancel culture” or used when people are harassing you or in mob vs. mob situations.

Here we use it as a way of adjusting what you see, similar to following… since SSB apps tend to show you neighbors in the social space in addition to people you’re actually following. It doesn’t necessarily have the same connotation of “I hate you and we’re enemies now”. Just “I don’t feel like reading your account”.

It’s also useful to avoid harassment, but that’s only one of its purposes.

What if
* _blocking_ --> **public mute** - a mute that others can observe
* _mute_ --> **private mute** - a mute that only you know about

It’s never going to feel great for someone to ignore you, but maybe it could feel softer.

People will arrive with their own ideas of “blocking” from the outside world so maybe it’s best to meet them where they’re at and use a different word for our different concept.

Or even a more radical change like
* _following_ --> **following**
* _no relationship_ --> **neighbors** or **no relationship** depending on the social graph
* _blocking_ --> **not reading**

For example people may want to see “neighbors” by default but not “no relationship” people.

This vocabulary needs some work.  Any thoughts?

**Comment by georgeowell** [comment]
I can really see where you are going with this and I like it but the thing is people are going to ask "How do I block someone?" and we'll be then be like "oh you should mute them". A lot of people expect "block" functionality.

For anyone reading who isn't familiar, the current UX is Patchwork is:
- Neutral
- Follow 
- Block

Then:
- Listen
- Ignore

I think "oh don't worry you can just mute that fascist" isn't going to cut it linguistically. I'm just searching in vain for another term for block :thinking: :thinking: :thinking:

**Comment by georgeowell** [comment]
Maybe we should think about principles of free listening with this. I’m contradicting what I said previously but:
- Listening
- Neighbour
- Not Listening

Or perhaps:
- Gossiping
- Neighbour
- Not Gossiping

**Comment by christianbundy** [comment]
I think there might be a deeper problem where public follows and blocks are the default and that has negative side-effects. My dream would be a network where all of your public relationships are strong, so public follows mean "I like and trust this person and I want the world to know" and public blocks mean "I don't like or trust this person and I want the world to know". I'm imagining a gradient like:

- **Public:** Trust (Vouch?)
- **Private:** Follow
- **Private:** Unfollow
- **Public:** Block (Flag?)

**Comment by georgeowell** [comment]
For some reason that got me thinking about myspace top friends :rofl:

**Comment by christianbundy** [comment]
New design mockup:

![myspace-notifications](https://user-images.githubusercontent.com/537700/73595975-e2f8d280-44d2-11ea-91b4-f840ccda8401.jpg)

But really though, I think there's some utility in differentiating between 'this is a cool account that I want to see' and 'this is a human I trust and want to share with my friends'. I guess that *does* have a similar effect to the top 8, but with the goal of building robust social fabric rather than ranking friends.

**Comment by georgeowell** [comment]
You're also verifying the crypto keys as well. It's a tangential security feature :smile:

**Comment by jedahan** [comment]
On facebook, at one point there was (and maybe still is) a little icon next to the post button. It shows a globe, and can be changed to the silhouettes of a few people, or I think maybe one person.

The long and short of it, is that it strongly tied the action you were about to perform with its expected visibility - public, friends, (and maybe private? not sure there is such thing on FB).

I think a lot of our settings and verbiage, are oasis-wide. Maybe we can consider giving dynamic controls at the point of action. I am not sure if this particular issue is an appropriate place to think about it.

**Comment by cinnamon-bun** [comment]
@georgeowell 
> I think "oh don't worry you can just mute that fascist" isn't going to cut it linguistically. 

It sounds like a weaker action and it is.  I don't want to give people a false sense of security that they've become invisible to the people they're blocking.  If that doesn't cut it for people then we need to add the technical capability to do the kind of blocking that people expect.

I wish we could write "friend-only posts" that only your friends could decrypt.  Not private groups, but everyone you're following.

@christianbundy 
Yeah it's awkward that "reading" and "trusting" are conflated in our following system.  They should be 2 independent variables...

**Comment by cinnamon-bun** [comment]
@georgeowell 
I like "Listening / Neighbor / Not Listening".  However it would be nice to differentiate between people near you in the social graph (neighbors?) and people far away (strangers?)

These words are all so loaded with associations.  I wonder if they'll prime people into thinking in certain ways.

"Gossip" sounds bidirectional but following is one-directional

**Comment by christianbundy** [comment]
>  I don't want to give people a false sense of security that they've become invisible to the people they're blocking.

I honestly completely forgot that "blocking" had the connotation "this person won't be able to see you". Huge +1 on removing the word "block" until we can provide some better privacy assurances. Maybe:

- **Public:** Public follow
- **Private:** Private follow
- **Private:** Private mute
- **Public:** Public mute

It's not perfect, but I think it's better.

**Comment by christianbundy** [comment]
Watching a demo from @arj03 of SSB-Browser, I think I really like how he does "Remove feeds". Maybe:

- Announce follow (public)
- Follow
- Neutral
- Remove
- Announce removal (public)

**Comment by cinnamon-bun** [comment]
@christianbundy 
To me "Remove" sounds like something that would apply to all viewers, like Ban or Kick, not a subjective action.

Big +1 from me on deleting data from blocked feeds.

If you remove the data of a blocked feed you'll also forget its name.  Do we need to store the names somewhere so we can show a list of blocked feeds?  Maybe it can go into the block message.

Hm, Forget is a nice word also

**Comment by christianbundy** [comment]
> Do we need to store the names somewhere so we can show a list of blocked feeds? Maybe it can go into the block message.

Yeah, or we can ask people to label feeds before they're deleted. :+1:
